internal/impl/kafka/enterprise/redpanda_migrator_offsets_input.go

```diff
@@ -0,0 +1,207 @@
+				Description("Sets the maximum amount of bytes a broker will try to send during a fetch. Note that brokers may not obey this limit if it has records larger than this limit. This is the equivalent to the Java fetch.max.bytes setting.").
+				Advanced().
+				Default("50MiB"),
+			service.NewStringField(rmoiFieldFetchMinBytes).
+				Description("Sets the minimum amount of bytes a broker will try to send during a fetch. This is the equivalent to the Java fetch.min.bytes setting.").
+				Advanced().
+				Default("1B"),
+			service.NewStringField(rmoiFieldFetchMaxPartitionBytes).
+				Description("Sets the maximum amount of bytes that will be consumed for a single partition in a fetch request. Note that if a single batch is larger than this number, that batch will still be returned so the client can make progress. This is the equivalent to the Java fetch.max.partition.bytes setting.").
+				Advanced().
+				Default("1MiB"),
+		},
+		kafka.FranzReaderOrderedConfigFields(),
+		[]*service.ConfigField{
+			service.NewAutoRetryNacksToggleField(),
+		},
+	)
+}
+
+func init() {
+	err := service.RegisterBatchInput("redpanda_migrator_offsets", redpandaMigratorOffsetsInputConfig(),
+		func(conf *service.ParsedConfig, mgr *service.Resources) (service.BatchInput, error) {
+			tmpOpts, err := kafka.FranzConnectionOptsFromConfig(conf, mgr.Logger())
+			if err != nil {
+				return nil, err
+			}
+			clientOpts := append([]kgo.Opt{}, tmpOpts...)
+
+			d := kafka.FranzConsumerDetails{}
+
+			var topics []string
+			if topicList, err := conf.FieldStringList(rmoiFieldTopics); err != nil {
+				return nil, err
+			} else {
+				topics, _, err = kafka.ParseTopics(topicList, -1, false)
+				if err != nil {
+					return nil, err
+				}
+				if len(topics) == 0 {
+					return nil, errors.New("at least one topic must be specified")
+				}
+			}
+
+			var topicPatterns []*regexp.Regexp
+			if regexpTopics, err := conf.FieldBool(rmoiFieldRegexpTopics); err != nil {
+				return nil, err
+			} else if regexpTopics {
+				topicPatterns = make([]*regexp.Regexp, 0, len(topics))
+				for _, topic := range topics {
+					tp, err := regexp.Compile(topic)
+					if err != nil {
+						return nil, fmt.Errorf("failed to compile topic regex %q: %s", topic, err)
+					}
+					topicPatterns = append(topicPatterns, tp)
+				}
+			}
+
+			if d.RackID, err = conf.FieldString(rmoiFieldRackID); err != nil {
+				return nil, err
+			}
+
+			if d.FetchMaxBytes, err = kafka.BytesFromStrFieldAsInt32(rmoiFieldFetchMaxBytes, conf); err != nil {
+				return nil, err
+			}
+			if d.FetchMinBytes, err = kafka.BytesFromStrFieldAsInt32(rmoiFieldFetchMinBytes, conf); err != nil {
+				return nil, err
+			}
+			if d.FetchMaxPartitionBytes, err = kafka.BytesFromStrFieldAsInt32(rmoiFieldFetchMaxPartitionBytes, conf); err != nil {
+				return nil, err
+			}
+
+			// Consume messages from the `__consumer_offsets` topic
+			d.Topics = []string{`__consumer_offsets`}
```

Comment:
What's the plan for Redpanda Serverless, which doesn't support this?

Reply:
I have a ticket for that and for now it's still something which needs to be discussed. I don't know what we can do there, but both the `redpanda_migrator_offsets` input and output don't currently work with RP Serverless.